backend-rust/src/indexer.rs

```diff
@@ -1293,298 +1176,353 @@ impl PreparedBlockItem {
-                AccountTransactionEffects::BakerAdded {
-                    data: event_data,
-                } => {
-                    let event = concordium_rust_sdk::types::BakerEvent::BakerAdded {
-                        data: event_data.clone(),
-                    };
-                    let prepared = PreparedBakerEvent::prepare(&event)?;
-                    PreparedEvent::BakerEvents(vec![prepared])
-                }
-                AccountTransactionEffects::BakerRemoved {
-                    baker_id,
-                } => {
-                    let event = concordium_rust_sdk::types::BakerEvent::BakerRemoved {
-                        baker_id: *baker_id,
-                    };
-                    let prepared = PreparedBakerEvent::prepare(&event)?;
-                    PreparedEvent::BakerEvents(vec![prepared])
-                }
-                AccountTransactionEffects::BakerStakeUpdated {
-                    data: update,
-                } => {
-                    let Some(update) = update else {
-                        return Ok(PreparedEvent::NoOperation);
-                    };
-                    let event = if update.increased {
-                        concordium_rust_sdk::types::BakerEvent::BakerStakeIncreased {
-                            baker_id:  update.baker_id,
-                            new_stake: update.new_stake,
-                        }
-                    } else {
-                        concordium_rust_sdk::types::BakerEvent::BakerStakeDecreased {
-                            baker_id:  update.baker_id,
-                            new_stake: update.new_stake,
                         }
-                    };
-                    let prepared = PreparedBakerEvent::prepare(&event)?;
-                    PreparedEvent::BakerEvents(vec![prepared])
-                }
-                AccountTransactionEffects::BakerRestakeEarningsUpdated {
-                    baker_id,
-                    restake_earnings,
-                } => PreparedEvent::BakerEvents(vec![PreparedBakerEvent::prepare(
+                    }
+                    _ => PreparedRejectedEvent::NoEvent,
+                };
+
+                PreparedEvent::RejectedTransaction(rejected_event)
+            }
+            AccountTransactionEffects::ModuleDeployed {
+                module_ref,
+            } => PreparedEvent::ModuleDeployed(
+                PreparedModuleDeployed::prepare(node_client, data, *module_ref).await?,
+            ),
+            AccountTransactionEffects::ContractInitialized {
+                data: event_data,
+            } => PreparedEvent::ContractInitialized(
+                PreparedContractInitialized::prepare(node_client, data, event_data, sender).await?,
+            ),
+            AccountTransactionEffects::ContractUpdateIssued {
+                effects,
+            } => PreparedEvent::ContractUpdate(
+                PreparedContractUpdates::prepare(node_client, data, effects).await?,
```

Comment:
Should the fees for making contract updates also be added here?

Reply:
They are all included in the transaction fee by the protocol, so we don't have to do anything